src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/PkcsPalWindows.Encrypt.cs

```diff
@@ -53,6 +58,38 @@ public sealed override byte[] Encrypt(CmsRecipientCollection recipients, Content
             }
         }
 
+        private static unsafe void ReencodeIfUsingIndefiniteLengthEncodingOnOuterStructure(ref byte[] encodedContent)
+        {
+            AsnReader reader = new AsnReader(encodedContent, AsnEncodingRules.BER);
+            Asn1Tag tag = reader.ReadTagAndLength(out int? contentsLength, out int _);
+
+            if (contentsLength != null)
+            {
+                // definite length, do nothing
+                return;
+            }
+
+            byte[] oldContent = encodedContent;
+
+            // Pin the content to prevent it from getting copied during heap compaction.
+            fixed (byte* pinnedOldContent = oldContent)
+            {
+                try
+                {
+                    using (AsnWriter writer = new AsnWriter(AsnEncodingRules.BER))
+                    {
+                        // Tag doesn't matter here as we won't write it into the document
+                        writer.WriteOctetString(reader.PeekContentBytes().Span);
+                        encodedContent = writer.Encode();
+                    }
+                }
+                finally
+                {
+                    Array.Clear(oldContent, 0, oldContent.Length);
```

Comment:
I know I said (in person) to switch to CryptographicOperations.ZeroMemory; but this file probably compiles under netstandard, which doesn't have that API.  So Array.Clear is fine.